### PR TITLE
[AL-17] Integrate register api

### PR DESCRIPTION
**Pull request by congtrung6391**
## Jira link

https://capstone-alumni.atlassian.net/browse/AL-17

## Describe your changes

Integrate register api

## Screenshot/Record (FE)
https://drive.google.com/file/d/1wxrAc4P9Lox_kcgWNfNEzN-gU-0N3GYH/view?usp=share_link

## Checklist before requesting a review
1. [x] I have performed a self-review of my code
2. [x] Have you lint your code locally before submission?

### CODE REVIEW

File: modules/sessions/components/SignUpForm.tsx [hunk 1]
```diff
@@ -24,61 +23,34 @@ import twitterFill from '@iconify/icons-eva/twitter-fill';
...
-  const router = useRouter();
+  const { signUp, isLoading } = useSignUp();
 
   const resolver = useYupValidateionResolver(validationSchema);
 
-  const { control, handleSubmit } = useForm<FormValues>({
+  const { control, handleSubmit } = useForm<SignUpFormValues>({
     mode: 'onChange',
     defaultValues: {
-      fullName: '',
```
Comment (congtrung6391) [inline modules/sessions/components/SignUpForm.tsx:81]: Considering: should we put full name in register form or in verification form

File: modules/sessions/hooks/useSignUp.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      method: 'POST',
+      url: '/api/signup',
+      data: values,
+    }),
+    {
+      revalidate: false,
+      rollbackOnError: false,
+      // TODO: sign-in before redirect
+      // TODO: add notification onError
+      onSuccess: () => router.push('/verify-account'),
```
Comment (congtrung6391) [inline modules/sessions/hooks/useSignUp.ts:22]: Expected:

1. user input info
2. click submit button
3. call register api
4. implicit sign-in
5. redirect to verify-account page
However, manual sign-in is not ready yet.

Comment (congtrung6391) [reply modules/sessions/hooks/useSignUp.ts:22]: We need something to notify user that there is an error.
Suggest: react-toasify

**Review by pvnamGit — APPROVED** [review]

**Comment by pvnamGit** [comment]
I don't see a error message notification in record

**Comment by congtrung6391** [comment]
> I don't see a error message notification in record

Screenshot 2022-12-28 202247

File: modules/sessions/hooks/useSignUp.ts (hunk 2 above)
Comment (congtrung6391) [reply modules/sessions/hooks/useSignUp.ts:22]: @pvnamGit we need to set up a toast or something like that, but it will be done in another task

**Pull request merged**